rfc/attestation-config.md

```diff
@@ -0,0 +1,102 @@
+    firmwareSignerConfig:
+      # List of accepted values for the firmware signing key digest.
+      acceptedKeyDigests:
+        - 7486a447ec0f1958002a22a06b7673b9fd27d11e1c6527498056054c5fa92d23c50f9de44072760fe2b6fb89740cc96
+      # Key digest enforcement policy. One of {'equal', 'maaFallback', 'warnOnly'}
+      #     - 'equal'       : Error if the reported signing key digest does not match any of the values in 'acceptedKeyDigests'
+      #     - 'maaFallback' : Use 'equal' checking for validation, but fallback to using Microsoft Azure Attestation (MAA) for validation if the reported digest does not match any of the values in 'acceptedKeyDigests'. See the Azure docs for more details: https://learn.microsoft.com/en-us/azure/attestation/overview#amd-sev-snp-attestation
+      #     - 'warnOnly'    : Same as 'equal', but only prints a warning instead of returning an error if no match is found
+      enforcementPolicy: maaFallback
+    # Lowest acceptable microcode version
+    microcodeVersion: 115
+    # Lowest acceptable SNP version
+    snpVersion: 8
+    # Lowest acceptable bootloader version
+    bootloaderVersion: latest
+```
+
+## In our code
+
+`/internal/config/` holds default values that will be written to the config file.
+
+`atls.Valdidator` initializers should accept a configuration struct which holds information about what values should be verified and how.
+Each Validator should provide default values that can imported by the config.
+
+```Go
+attestCfg := sevcfg.New()
+cfg.Measurements = measurements
+validator := sev.NewValidator(attestCfg, log)
+```
+
+A new method `*Config.AttestationConfig()` returns an interface of the attestation configuration.
+The interface should provide a `*Cfg.Variant()` method to retrieve the attestation variant used by the cluster.
+This should work similar to how we retrieve the CSP using `*Config.GetProvider()`.
+
+Creation of a Validator should require using a concrete type, so the attestation config interface needs to be cast to the configuration required by the validator.
+
+Additional methods to set/retrieve common values of the config interface may be defined, for example to set expected measurements,
+since these are currently used by all attestation variants.
+
+If an attestation config specifies the minimum version of a parameter as `latest`,
+that value is substituted with the most recent version of that parameter for the given CSP from our API.
+The value substitution is part of the unmarshalling logic.
+
+## Attestation config API
+
+Config values are uploaded to S3 and can be accessed via HTTP.
+
+The attestation config API uses the same CSP names as [the image API](./image-api.md#image-api-endpoints).
+
+The following HTTP endpoint is available:
+
+- `GET /constellation/v1/attestation/<ATTESTATION_VARIANT>/`
+  - `latest` returns the latest configuration available for a given attestation variant, as well as a reference to all past configurations
```

Comment:
Imo it shouldn't give you a list if it is called latest. This will lead to much confusion with the other parts of our api. Why not just call it 'list' and return all versions? As long as they sort (what they should), the caller can find the latest version herself.